### PR TITLE
Adding 2 new messages 0x8012 and 0x9999

### PR DESCRIPTION
**Pull request by pipiche38**

### CODE REVIEW

**Comment by pipiche38** [comment]
@doudz , don't where the changes on Application.py are coming from.

File: zigpy_zigate/api.py [hunk 1]
```diff
@@ -29,14 +29,17 @@
     0x8007: (t.uint8_t,),
     0x8009: (t.NWK, t.EUI64, t.uint16_t, t.uint64_t, t.uint8_t),
     0x8010: (t.uint16_t, t.uint16_t),
+    0x8012: (t.uint8_t, t.uint8_t, t.uint8_t, t.Address, t.uint8_t),
```
Comment (doudz) [inline zigpy_zigate/api.py:32]: Duplicated line

File: zigpy_zigate/zigbee/application.py [hunk 2]
```diff
@@ -106,17 +108,18 @@ def zigate_callback_handler(self, msg, response, lqi):
             nwk = response[0]
             ieee = zigpy.types.EUI64(response[1])
             parent_nwk = 0
-            self.handle_join(nwk, ieee, parent_nwk)
+            #time.sleep(2.0)
```
Comment (doudz) [inline zigpy_zigate/zigbee/application.py:111]: Please remove commented line

File: zigpy_zigate/zigbee/application.py [hunk 3]
```diff
@@ -106,17 +108,18 @@ def zigate_callback_handler(self, msg, response, lqi):
...
+            #self.handle_join(nwk, ieee, parent_nwk)
```
Comment (doudz) [inline zigpy_zigate/zigbee/application.py:112]: same

**Review by doudz — CHANGES_REQUESTED** [review]
Could you also add a unit test to validate two stages pairing ?

File: zigpy_zigate/zigbee/application.py [hunk 4]
```diff
@@ -106,17 +108,18 @@ def zigate_callback_handler(self, msg, response, lqi):
...
-            # rejoin = response[3]
-            # if nwk in self._pending_join or rejoin:
-            #     LOGGER.debug('Finish pairing {} (2nd device announce)'.format(nwk))
-            #     if nwk in self._pending_join:
-            #         self._pending_join.remove(nwk)
-            #     self.handle_join(nwk, ieee, parent_nwk)
-            # else:
-            #     LOGGER.debug('Start pairing {} (1st device announce)'.format(nwk))
-            #     self._pending_join.append(nwk)
+            rejoin = response[3]
```
Comment (doudz) [inline zigpy_zigate/zigbee/application.py:114]: You should also update line 47 to set minimal firmware to 3.2

**Review by doudz — CHANGES_REQUESTED** [review]

File: zigpy_zigate/api.py [hunk 5]
```diff
@@ -29,14 +29,17 @@
...
     0x8011: (t.uint8_t, t.NWK, t.uint8_t, t.uint16_t, t.uint8_t),
+    0x8012: (t.uint8_t, t.uint8_t, t.uint8_t, t.Address, t.uint8_t),
     0x8017: (t.uint32_t,),
     0x8024: (t.uint8_t, t.NWK, t.EUI64, t.uint8_t),
     0x8035: (t.uint8_t, t.uint32_t),
     0x8048: (t.EUI64, t.uint8_t),
     0x8701: (t.uint8_t, t.uint8_t),
     0x8702: (t.uint8_t, t.uint8_t, t.uint8_t, t.Address, t.uint8_t),
     0x8806: (t.uint8_t,),
+    0x9999: (t.uint8_t,),
```
Comment (doudz) [inline zigpy_zigate/api.py:42]: You should also handle the message with logger

**Review by doudz — CHANGES_REQUESTED** [review]

**Comment by pipiche38** [comment]
Ok. will leave @badz handling.

**Comment by pipiche38** [comment]
we are currently working on zigpy-zigate library under a fork dedicated to our project, we don't have the possibility to test it under the ZHA environment but if you are ready to do, you can eventually use it
https://github.com/zigbeefordomoticz/zigpy-zigate

When ready we will be providing PRs against the main stream

**Pull request closed without merging**